### DISCUSSION TITLE
Getting Started with fhir.resources and Synthea

### DISCUSSION THREAD

**Discussion by fdefalco** [nazrulworld/fhir.resources#136] [Q&A] [answered]
I'm brand new to using this library so apologies if I'm missing something basic.  I'm generating sample FHIR bundles using the [Synthea](https://github.com/synthetichealth/synthea) project.  I've attached my sample file.  When attempting to parse the file it fails with 2067 validation errors.  I've attached the sample file. 

The code is very simple.

```python
import os

from fhir.resources.patient import Patient
from fhir.resources.bundle import Bundle

patientFile = "Agripina633_Howe413_8aca2c5e-a829-09e1-74f2-d15cec8a9008.json"
patientBundle = Bundle.parse_file(patientFile)
```

[Agripina633_Howe413_8aca2c5e-a829-09e1-74f2-d15cec8a9008.json.zip](https://github.com/nazrulworld/fhir.resources/files/12110537/Agripina633_Howe413_8aca2c5e-a829-09e1-74f2-d15cec8a9008.json.zip)

I'm curious if this error is due to my usage or version of the simulated FHIR or something else?  The error with the first few validation errors are as follows:

patientBundle = Bundle.parse_file(patientFile)
File "/opt/homebrew/lib/python3.11/site-packages/fhir/resources/core/fhirabstractmodel.py", line 243, in parse_file
return cls.parse_obj(obj)
File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
File "/opt/homebrew/lib/python3.11/site-packages/fhir/resources/core/fhirabstractmodel.py", line 105, in __init__
BaseModel.__init__(__pydantic_self__, **data)
File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 2067 validation errors for Bundle
entry -> 1 -> resource -> address
  extra fields not permitted (type=value_error.extra)
entry -> 1 -> resource -> telecom
  extra fields not permitted (type=value_error.extra)
entry -> 2 -> resource -> telecom
  extra fields not permitted (type=value_error.extra)
entry -> 4 -> resource -> telecom
  extra fields not permitted (type=value_error.extra)
entry -> 5 -> resource -> class
  value is not a valid list (type=type_error.list)

**Comment by nazrulworld** [comment]
Hi @fdefalco 
Sorry for late response, I was in vacation. I am coming back to you ASAP.

**Comment by nazrulworld** [comment, marked as answer]
@fdefalco I think you were using wrong FHIR version, perhaps the bundle you attached is FHIR version R4/R4B.
`from fhir.resources.R4B.bundle import Bundle`  
`patientBundle = Bundle.parse_file(patientFile)`

**Comment by fdefalco** [comment]
Thanks, sorry for missing a simple resolution.  Is there no way to infer the FHIR version from the document itself so I might allow my data pipeline to be more resilient and use the proper bundle version based on the content of the file?
